src/main/java/com/suse/saltstack/netapi/config/ClientConfig.java

```diff
@@ -12,6 +12,25 @@
     public static final Key<URI> URL = new Key<>(URI.create("http://localhost:8000"));
     public static final Key<String> TOKEN = new Key<>();
 
+    // Request timeout settings
+    /**
+     * Sets a specified timeout value, in milliseconds, to be used when dealing a communications link in HttpClientConnection / JDKConnection.
+     * A timeout of zero is interpreted as an infinite timeout. Default value is 10000ms (10s).
+     * 
+     * @see HttpClientConnection.request(String)
+     * @see JDKConnection.request(String, String)
+     */
+    public static final Key<Integer> CONNECT_TIMEOUT = new Key<>(10000);
+
+    /**
+     * Sets a specified timeout value, in milliseconds, to be used when waiting for data in HttpClientConnection / JDKConnection.
+     * A timeout of zero is interpreted as an infinite timeout. Default value is 10000ms (10s).
```

Comment:
And here too:
`Timeout in milliseconds for waiting for data. A timeout of zero is interpreted as an infinite timeout. Default value is 10000ms (10s)`